client/src/components/Sub_ProductDetail/ImageGallery.jsx

```diff
@@ -25,18 +26,47 @@ function ImageGallery({ currentStyle, productID }) {
     }
   }, [currentStyle]);
 
+  useEffect(() => {
+    if (showModalView !== 'default') document.body.style.overflow = 'hidden';
+    else document.body.style.overflow = 'scroll';
+  }, [showModalView]);
+
   return (
+    // <div className={`image-gallery ${showExpandView ? 'expand-view' : ''}`}>
     <div className="image-gallery">
-      <ImageSlide
-        images={currentStyle.photos}
-        currentImageIndex={currentImageIndex}
-        setCurrentImageIndex={setCurrentImageIndex}
-      />
-      <DefaultView
-        images={currentStyle.photos}
+      {showModalView === 'expand'
+    && (
+    <ModalView
+      modalClass="expand-view"
+      currentStyle={currentStyle}
+      currentImageIndex={currentImageIndex}
+      setCurrentImageIndex={setCurrentImageIndex}
+      showModalView={showModalView}
+      setShowModalView={setShowModalView}
+    />
+    )}
+
+      {showModalView === 'zoom'
+    && (
+    <ModalView
+      modalClass="zoom-view"
+      currentStyle={currentStyle}
+      currentImageIndex={currentImageIndex}
+      setCurrentImageIndex={setCurrentImageIndex}
+      showModalView={showModalView}
+      setShowModalView={setShowModalView}
+    />
+    )}
+
```

Comment:
beautiful, you have a modular modal view class. this is exactly what i wanted to implement. Can't wait to see how you did it.